pkg/collectors/pod.go

```diff
@@ -256,6 +268,8 @@ func (pc *podCollector) Describe(ch chan<- *prometheus.Desc) {
 	ch <- descPodContainerResourceLimitsNvidiaGPUDevices
```

Comment:
the descriptions should be conditional as well

Reply:
Will do.

Reply:
I have tested this. It will not produce any old metrics even we add them to the channel. But i think it will be more clear and safe to make them conditional.